content/tracing/setup/first_class_dimensions.md

```diff
@@ -0,0 +1,149 @@
+* Other characters must be alphanumeric lower case Unicode characters, underscores, minuses, colons, periods or slashes.
+* They must not be more than 100 characters long.
+
+Environments in traces and configuration files are normalized:
+
+* Unsupported characters are replaced by underscores.
+* Upper case characters are converted to lower case.
+
+### Default environment
+
+The default environment for un-tagged data is `env:none`. See below to see how to specify custom environments:
+
+### Setup
+
+There are several ways to specify an environment when reporting data:
+
+1. Host tag:  
+  Use a host tag with the format `env:XXXX` to tag all traces from that Agent accordingly.
+
+2. Agent configuration:  
+  Override the default tag used by the trace Agent in [the Agent configuration file][2]. This tags all traces coming through the Agent, overriding the host tag value.
+
+    ```
+    [trace.config]
+    env = pre-prod
+    ```
+
+3. Per trace:  
+  When submitting a single trace, specify an environment by tagging one of its spans with the metadata key `env`. This overrides the Agent configuration and the host tags value (if any).  
+
+  * **Go**:
+    ```
+    tracer.SetMeta(“env”, “prod”)
+    ```
+  For OpenTracing use the “GlobalTags” field in the “opentracing.Configuration” structure.
+
+  * **Java**:  
+      Via sysprop: 
+      ```
+      -Ddd.trace.span.tags=”env:prod”
+      ```
+      Via env var:
+      ```
+      DD_TRACE_SPAN_TAGS=”env:prod”
+      ```
+
+  * **Ruby**:
+  ```
+  Datadog.tracer.set_tags(‘env’ => ‘prod’)
+  ```
+
+  * **Python**:
+    ```
+    from ddtrace import tracer
+    tracer.set_tags('env', 'prod')
+    ```
+
+### Viewing Data by Environment
+
+Environments appear near the top of APM pages. Use the dropdown to scope the data displayed on the current page.
+
+{{< img src="tracing/setup/first_class_dimensions/envs_tracing_screen.png" alt="Envs tracing" responsive="true" popup="true" style="width:80%;">}}
+
+
+## Primary Tags
+### Definition
+
+A primary tag is a first-class dimension that you use to scope an entire Datadog APM application. Primary tags can be used in conjunction with environments to get an even finer view of your application’s behavior. Use cases for primary tags include:
+
+* availability zone
+* datacenter
+
+You can optionally set 1 primary tag. A primary tag must follow these rules:
+
+* It must start with a lowercase letter.
+* Other characters must be alphanumeric lowercase Unicode characters, underscores, minuses, colons, periods, or slashes.
+* It must not be more than 100 characters long.
+
+### Setup
+APM primary tags must be set up in two ways: in the trace Agent and in the Datadog UI.
+
+#### Agent configuration
+Override the default tag used by the trace Agent in the Agent configuration file. This tags all traces coming through the Agent, overriding the host tag value. An APM primary tag is configured like any regular host tag. You can specify it as:
```

Comment:
Let's hyperlink "Agent configuration file" here like we do above:
https://docs.datadoghq.com/agent/basic_agent_usage/#configuration-file